sc-memory/sc-memory/sc_agent_manager.tpp

```diff
@@ -111,21 +124,27 @@ void ScAgentManager<TScAgent>::Unsubscribe(
   agent.SetInitiator(context->GetUser());
   agent.SetImplementation(agentImplementationAddr);
 
-  std::string const & agentName = agent.GetName();
-
-  auto const & eventClassIt = ScAgentManager<TScAgent>::m_agentEventClasses.find(agentName);
-  if (eventClassIt != ScAgentManager<TScAgent>::m_agentEventClasses.cend()
-      && eventClassIt->second == ScKeynodes::sc_event_before_erase_element)
+  std::string const & agentClassName = agent.GetName();
+  if (IsAgentSubscribedToEventOfErasingElement(agentClassName))
```

Comment:
someone may subscribe agent for erasing of element and then after some logic unsubscribe it(for example during  transaction), why is this an exception?

Reply:
In either case, the agent's unsubscription from event of erasing sc-element happens by itself

Reply:
it is possible that element is not erased but subscription to that event should be removed

Reply:
Fixed